app-misc/hello/hello-2.10.recipe

```diff
@@ -0,0 +1,49 @@
+SUMMARY="GNU Hello program"
+DESCRIPTION="The GNU Hello program produces a familiar, friendly greeting.\
+Yes, this is another implementation of the classic program that prints \"Hello, world!\" when you run it.\
+However, unlike the minimal version often seen, GNU Hello processes its argument list to modify its behavior,\
+supports greetings in many languages, and so on.\
+The primary purpose of GNU Hello is to demonstrate how to write other programs that do these things;\
+it serves as a model for GNU coding standards and GNU maintainer practices."
+HOMEPAGE="https://www.gnu.org/software/hello/"
+COPYRIGHT="1992-2014 Free Software Foundation, Inc."
+LICENSE="GNU GPL v3"
+REVISION="2"
+SOURCE_URI="http://ftp.gnu.org/gnu/hello/hello-$portVersion.tar.gz"
+CHECKSUM_SHA256="31e066137a962676e89f69d1b65382de95a7ef7d914b8cb956f41ea72e0f516b"
+
+ARCHITECTURES="x86 x86_gcc2 x86_64"
+
+PROVIDES="
+	hello = $portVersion
+	cmd:hello = $portVersion
+	"
+REQUIRES="
+	haiku
+	gnulib
```

Comment:
It looks like the hello program does not need gnulib.
```diff
 REQUIRES="
 	haiku
- 	gnulib
 	lib:libintl
 	"
```

And it does not seem to be required at run time.
Well, maybe this is an optional build dependency. In that case we can add this:
```diff
 BUILD_PREREQUIRES="
 	cmd:autoconf
 	cmd:gcc
 	cmd:gettext
+	cmd:gnulib_tool
 	cmd:help2man
 	cmd:make
 	"
```

Let's see what the other contributors say about all this.

Reply:
i've found about this dep into the README.dev. Is there a better way to check for dependencies (and the minimum version)?

Reply:
No need for the gnulib_tool :)

Reply:
> I've found about this dep into the README.dev. Is there a better way to check for dependencies (and the minimum version)?

I'd say that the README-dev is a good start. Then, if I suspect that a dependency is not really needed, I try to build the package without that dependency and check if it is still building fine and if the resulting package has the same contents. To compare the contents I use "package extract ..." and then make sure that every runtime (or library) has the same checksum.